### PR TITLE
LPS-74408 Add metal js component to add fragment entries view

### PR DESCRIPTION
**Pull request by p2kmgcl**

### CODE REVIEW

**Comment by p2kmgcl** [comment]
ci:retest

**Comment by p2kmgcl** [comment]
ci:retest

@carloslancha ya está usando metal-modal :)

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/build.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 dependencies {
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/build.gradle]: @carloslancha do I move from shrinkwrap to package-lock?

**Comment by p2kmgcl** [comment]
@carloslancha done :smile:

**Comment by p2kmgcl** [comment]
ci:retest

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentNameEditor.soy [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+						</span>
+						<span class="hide-accessible">
+							{msg desc=""}required{/msg}
+						</span>
+					</label>
+					<input class="field form-control {if $error}error-field{/if}" name="{$namespace}name" type="text">
+					{if $error}
+						<div class="form-validator-stack help-block">
+							<div role="alert">
+								{if $error == 'class com.liferay.modern.site.building.fragment.exception.MSBFragmentEntryNameException'}
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentNameEditor.soy]: Can't backend send you an understable error?

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentNameEditor.soy [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+									{msg desc=""}this-name-already-exists-please-try-another-one{/msg}
+								{else}
+									{msg desc=""}unknown-error{/msg}
+								{/if}
+							</div>
+						</div>
+					{/if}
+				</div>
+				<div class="pull-right">
+					<button class="btn btn btn-cancel btn-default btn-link" type="button" data-onclick="{$_handleCancelButtonClick}">
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentNameEditor.soy]: Use Modal `hide` event and propagate it to the parent

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditor.es.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	shouldUpdate(changes) {
+		return changes && changes._aceEditor && changes._aceEditor.newVal;
+	}
+
+	/**
+	 * Callback executed when the ace editor is ready to use
+	 * @param {{editor: object}} Editor received inside the event
+	 */
+	_handleEditorReady({aceEditor}) {
+		this._aceEditor = aceEditor;
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditor.es.js]: Don't needed

**Comment by p2kmgcl** [comment]
@carloslancha done :), I'm reordering these commits and closing the pr

**Comment by carloslancha** [comment]
@p2kmgcl just send it to Eudaldo and cc me. Thx.

**Comment by p2kmgcl** [comment]
https://github.com/ealonso/liferay-portal/pull/1105

**Pull request closed without merging**